examples/platform/silabs/SiWx917/BUILD.gn

```diff
@@ -292,6 +294,11 @@ source_set("siwx917-common") {
     deps += [ ":siwx917-matter-shell" ]
   }
 
+  # DIC
+  if (enable_dic) {
+    deps += [ "${examples_plat_dir}:silabs-dic" ]
```

Comment:
should be changed to public_dep. see explanation in my other comment below

Reply:
@jmartinez-silabs  Modified DIC part as public_dep and those changes are part of efr32 and soc Build.gn in this PR.